en/companion_computer/auterion_skynode.md

```diff
@@ -0,0 +1,66 @@
+
+The onboard software is Auterion OS, consisting of an enterprise-hardened version of PX4 running on the flight controller, and an operating system with advanced management software running on the mission computer.
+The OS is managed by Auterion in production, with customer applications running as "add ons" in a safe sandbox within the mission computer.
+
+Auterion OS and Skynode allow seamless integration with Auterion's other software and fleet management products.
+
+For information about Auterion and Skynode, including purchase links:
+
+- [auterion.com](https://auterion.com/)
+- [Skynode](https://auterion.com/product/skynode/) (auterion.com)
+- Skynode Guides:
+  - [Manufacturer's Guide](https://docs.auterion.com/manufacturers/getting-started/readme) 
+  - [App Developer's Guide](https://docs.auterion.com/developers/getting-started/readme)
+
+
+## Skynode with Vanilla PX4
+
+Skynode comes with a managed release of PX4.
+If you would like to try a more recent PX4 flight kernel, you can install the upstream "vanilla" PX4 from [PX4/PX4-Autopilot](https://github.com/PX4/PX4-Autopilot).
+
+Upstream PX4 will generally work, with the following caveats:
+
+- Configuration for your exact product may not be present.
+  You may lose the configuration for ESCs, batteries, sensor configuration, and so on.
+- Some parameters might have different default values in the PX4 release shipped with Auterion OS.
+- Features accessed by vendor-specific customisations running on the companion computer may not be present in PX4.
+
+
+## Building/Uploading Firmware
+
+PX4 `px4_fmu-v5x` binaries for Skynode are built from source using the normal [developer environment](../dev_setup/dev_env.md) and [build commands](../dev_setup/building_px4.md), and are uploaded using either `upload_skynode_usb` or `upload_skynode_wifi` upload targets.
+
+You will need to enter Skynode's developer image password (twice) during the upload process.
+
+To build and upload PX4 for Skynode [when connected by USB-C](https://docs.auterion.com/manufacturers/avionics/skynode/advanced-configuration/connecting-to-skynode):
+
+```
+make px4_fmu-v5x upload_skynode_usb
+```
+
+To build and upload PX4 for Skynode when connected using [Skynode's WiFi network](https://docs.auterion.com/manufacturers/avionics/skynode/advanced-configuration/configuration):
+
+```
+make px4_fmu-v5x upload_skynode_wifi
+```
+
+:::note
+`upload_skynode_usb` and `upload_skynode_wifi` connect to Skynode via SSH over a network interface using the default (fixed) IP addresses, and upload a TAR compressed binary to the mission computer.
+The mission computer then decompresses the binary and installs it to the flight controller.
+
+SSH and TAR are upload dependencies, but are expected to be present by default on Ubuntu, Ubuntu running on Windows in WSL2, and macOS.
+:::
+
+## Restoring the Default PX4 Firmware
+
+To restore PX4:
+
+1. ?
```

Comment:
This is the quickest way to revert I've found:
```
ssh auterion@10.41.1.1
cp /persistent/shared_container_dir/fmu/update.tar /persistent/shared_container_dir/fmu/update-dev.tar
tail --follow=name /shared_container_dir/fmu/update_status
```
Does this need to be in the script as well? I can see if that's doable.

Reply:
I don't think this needs to be in the script - in particular because the locations of the dev image might not be in a standard location.

I'm going to copy this in, then add some questions for you.